ssl/ssl_cert.cc

```diff
@@ -415,6 +415,20 @@ bool ssl_has_certificate(const SSL_HANDSHAKE *hs) {
          ssl_has_private_key(hs);
 }
 
+bool ssl_has_certificate_pha(const SSL *ssl) {
+  assert(!ssl->server);
+  if (!ssl_cert_check_cert_private_keys_usage(ssl->s3->pha_config->client_cert.get())) {
+    return false;
+  }
+
+  CERT_PKEY &cert_pkey =
+      ssl->s3->pha_config->client_cert
+          ->cert_private_keys[ssl->s3->pha_config->client_cert->cert_private_key_idx];
+  return cert_pkey.chain != nullptr &&
```

Comment:
When I follow the chain of calls up to see how error is handled (like this null chain), there's very little information provided to the client about what went wrong. 

Here, and elsewhere that errors occur, are there appropriate (`OPENSSL_PUT_ERROR`) errors that could be set, or in the chain of callers above the location, that would provide more information to the client about a failed connection?